scalalib/src/mill/scalalib/JavaModule.scala

```diff
@@ -159,6 +194,109 @@ trait JavaModule
    */
   def runIvyDeps: T[Agg[Dep]] = Task { Agg.empty[Dep] }
 
+  def parentDep: T[Option[Dep]] = Task { None }
```

Comment:
Let's call this `bomParentDep`? To make it clear what it's used for

Reply:
It's not only used as a BOM dependency: its value is used from `PublishModule`, so that it ends up advertized as "parent POM" in the POM. `parentDep` conveys that dual use better IMO.

Reply:
I not convinced we need a `parentDep`. It's mapped to something Maven specific Mill doesn't even support. Can you elaborate why we need it and what it does exactly?

Reply:
I removed `parentDep`, given it doesn't bring much.

My initial idea was to complete early support for parent stuff added in https://github.com/com-lihaoyi/mill/pull/3756 (it adds `PublishModule#pomParentProject` in particular). But we might as well make Maven import add the parent to `bomDeps` alongside setting `pomParentProject`.

Reply:
Probably off-topic, about parent: I don't think we should model any parent-pom semantic. If we migrate from Maven, we should either ignore or somehow map to dependency imports, if possible. The addition of anything supporting "parent" to `PublishModule` doesn't look right to me. Since Mill does not have parents, we also don't need to generate `pom.xml` files referencing parents for publishing.

Reply:
I agree that there should be no need to model parent POM semantics in Mill. I don't know what motivated it in https://github.com/com-lihaoyi/mill/pull/3756